### ISSUE TITLE
try/except block in datagrepper.runner is overzealous

### ISSUE THREAD

**Issue by ianweller** [fedora-infra/datagrepper#76]
Running into an issue now on datagrepper01.stg where this traceback is happening:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/twisted/python/context.py", line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/lib64/python2.6/site-packages/twisted/python/context.py", line 37, in callWithContext
    return func(*args,**kw)
  File "/usr/lib64/python2.6/site-packages/twisted/internet/selectreactor.py", line 146, in _doReadOrWrite
    why = getattr(selectable, method)()
  File "/usr/lib/python2.6/site-packages/txzmq/connection.py", line 241, in doRead
    log.callWithLogger(self, self.messageReceived, message)
--- <exception caught here> ---
  File "/usr/lib64/python2.6/site-packages/twisted/python/log.py", line 84, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/usr/lib64/python2.6/site-packages/twisted/python/log.py", line 69, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/usr/lib64/python2.6/site-packages/twisted/python/context.py", line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/lib64/python2.6/site-packages/twisted/python/context.py", line 37, in callWithContext
    return func(*args,**kw)
  File "/usr/lib/python2.6/site-packages/txzmq/pubsub.py", line 60, in messageReceived
    self.gotMessage(message[1], message[0])
  File "/usr/lib/python2.6/site-packages/moksha/hub/zeromq/zeromq.py", line 168, in chain_over_moksha_callbacks
    f(_body, _topic)
  File "/usr/lib/python2.6/site-packages/moksha/hub/zeromq/zeromq.py", line 195, in intercept
    return callback(ZMQMessage(_topic, _body))
  File "/usr/lib/python2.6/site-packages/moksha/hub/api/consumer.py", line 107, in _consume_json
    self._consume(message_as_dict)
  File "/usr/lib/python2.6/site-packages/fedmsg/consumers/__init__.py", line 140, in _consume
    self.consume(message)
  File "/usr/lib/python2.6/site-packages/datagrepper/runner.py", line 42, in consume
    'datagrepper_{0}'.format(job.id))
  File "/usr/lib/python2.6/site-packages/moksha/hub/api/consumer.py", line 90, in _consume_json
    topic = message.headers[0].routing_key
exceptions.AttributeError: 'ZMQMessage' object has no attribute 'headers'
```

As far as I can tell, the `AttributeError` is occurring in another thread, but it's being caught by this runner. :(

**Comment by ralphbean** [comment]
No longer needed after #158.
